interop-cats/shared/src/main/scala/zio/interop/stm/STM.scala

```diff
@@ -220,7 +220,7 @@ object STM {
   final def succeed[F[+_], A](a: A): STM[F, A] = new STM(ZSTM.succeed(a))
 
   final def succeedLazy[F[+_], A](a: => A): STM[F, A] =
-    new STM(ZSTM.succeedLazy(a))
+    new STM(ZSTM.succeed(a))
```

Comment:
Ought to deprecate AND preserve laziness with `ZSTM.unit.as(a)`

Reply:
Actually `succeed` is already lazy

Reply:
Err, no, `a` is not by-name, it will evaluate `a` when applied, not when evaluated. https://github.com/zio/zio/blob/master/core/shared/src/main/scala/zio/stm/STM.scala#L766
You could say that `ZSchedule.succeedLazy` is already strict because it immediately evaluates when calling `succeed` (a change in this PR that went unnoticed https://github.com/zio/zio/pull/1377). Actually, I think just putting a deprecation here should be enough (you can call deprecated methods inside deprecated methods without a warning AFAIK)

Reply:
Oops, my bad. OK I'll keep calling the deprecated one since this one is deprecated.